### PR TITLE
refactor(engine): add moveRelative command to engine

### LINKED ISSUE #8424: HTTP API: Add a relative-move (jog) command
## Overview

As a robot client, I'd like to instruct the robot to move a relative amount in a given direction. Combined with #8425, this gives me a rudimentary interface to use the OT-2 as a "ruler".

## Acceptance criteria

- [ ] A ProtocolEngine/Session accepts a `moveRelative` command
    - Takes an "axis" and a distance

## Implementation details

**Command proposal**:

```ts
{
    commandType: "moveRelative",
    data: {
        pipetteId: string,    // pipette to move
        axis: "x", "y", "z",  // axis to move on
        distance: number,     // distance to move (negative allowed)
    },
    result: {}
}
```

### PR DESCRIPTION
**Pull request by mcous**
## Overview

This PR adds a `moveRelative` command to the engine. Closes #8424 

## Changelog

- add moveRelative command to engine
- change type of `DeckSlotName` enum to `str` to match JSON protocol schema

## Review requests

- [ ] Try this out on a real robot via the HTTP API

## Risk assessment

Very low

### CODE REVIEW

**Review by shlokamin — APPROVED** [review]
🚢

File: api/src/opentrons/protocol_engine/execution/movement.py [hunk 1]
```diff
@@ -80,8 +82,29 @@ async def move_to_well(
...
+            pipette_id=pipette_id,
+        )
+        hw_mount = pipette_location.mount.to_hw_mount()
+        delta = Point(
+            x=distance if axis == MovementAxis.X else 0,
+            y=distance if axis == MovementAxis.Y else 0,
+            z=distance if axis == MovementAxis.Z else 0,
+        )
+
+        await self._hardware_api.move_rel(mount=hw_mount, delta=delta)
```
Comment (sanni-t) [inline api/src/opentrons/protocol_engine/execution/movement.py]: `_hardware_api.move_rel` has a bug where if the current position of a mount is unknown then it'll home the robot, but somewhere down the line forget to update the current position so it does relative distance measurement from (x=0, y=0). So the delta we pass here for x or y becomes absolute deck coordinate which the pipette tries to go to.

**Review by sanni-t — APPROVED** [review]
Tested on a robot. Works perfectly as long as the robot is homed before issuing the command. 
We could do a check here for whether gantry has been homed before doing the relative move. But can be punted to later for a fix in hardware api.

**Comment by mcous** [comment]
To follow up on @sanni-t's observations, (we think) due to the initial z-stage home upon boot, the value of `hardware_api._current_position` once the application is usable is:

```python
{<Axis.X: 0>: 0.0, <Axis.Y: 1>: 0.0, <Axis.A: 3>: 170.15, <Axis.Z: 2>: 218.0, <Axis.B: 4>: 0.0, <Axis.C: 5>: 0.0}
```

Meanwhile, `smoothie_driver.homed_flags` looks like:

```python
{'X': False, 'Y': False, 'Z': True, 'A': True, 'B': False, 'C': False}
```

This is bad! Neither X, Y, B, nor C is homed, but the hardware controller logic, which simply looks for a truthy value of `self._current_position` to determine homing status, thinks the robot is all homed

**Pull request merged**